controllers/controller.go

```diff
@@ -224,23 +236,60 @@ func (i instanceReconcilerHelper) reconcileInstance(ctx context.Context, o clien
 	return ctrl.Result{}, nil
 }
 
-func (i instanceReconcilerHelper) checkPreconditions(o client.Object) (bool, ctrl.Result, error) {
-	i.rec.Event(o, corev1.EventTypeNormal, eventWaitingforPreconditions, "waiting for preconditions of the instance")
+func (i instanceReconcilerHelper) checkPreconditions(ctx context.Context, o client.Object) (bool, error) {
+	i.rec.Event(o, corev1.EventTypeNormal, eventWaitingForPreconditions, "waiting for preconditions of the instance")
+
+	// Checks references
+	if oRef, ok := o.(refsObject); ok {
+		i.log.Info("checking resource references")
+		ready, err := i.checkRefs(ctx, oRef)
+		if err != nil {
+			i.log.Info(fmt.Sprintf("one or more references can't be found yet: %s", err))
+			return true, err
```

Comment:
It doe's not make sense here since reconciliation will error and never re-queue. We need to keep the original function signature to allow this. After many errors, the k8s control plane puts resource reconciliation on halt for some significant time, so it is not a good idea to make an error without a critical reason.

Reply:
By simply fixing this code we will get:
```
--- PASS: kuttl (480.41s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/projectvpc-ref (425.37s)
```
So boost the e2e test perfonca from 21 minute to 8

Reply:
Nice catch! This is definitely should not return true.

Reply:
Nice catch, thanks, should return nil.